source/particle/world.cc

```diff
@@ -85,19 +85,26 @@ namespace aspect
     std::string
     World<dim>::generate_output() const
     {
+      std::vector<std::string> filenames;
+
       // If we do not write output
       // return early with the number of particles that were advected
-      if (!output)
+      if (output.empty())
         return "";
 
       TimerOutput::Scope timer_section(this->get_computing_timer(), "Particles: Output");
       const double output_time = (this->convert_output_to_years() ?
                                   this->get_time() / year_in_seconds :
                                   this->get_time());
 
-      const std::string filename = output->output_particle_data(particles,
-                                                                property_manager->get_data_info(),
-                                                                output_time);
+      std::string filename;
+
+      for ( typename std::vector<std_cxx11::shared_ptr<Output::Interface<dim> > >::const_iterator itr= output.begin(); itr != output.end(); itr++)
+        {
+          filename = (itr->get()->output_particle_data(particles,
```

Comment:
Since `output` is a vector of shared pointers, we dereference once to a shared pointer to output interface, and must dereference again to access the output interface object. `itr->get()->` is necessary.